core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -287,7 +262,81 @@ public void messageReceived(final Request request, final TransportChannel channe
 
         @Override
         public void messageReceived(Request request, TransportChannel channel, Task task) throws Exception {
-            new PrimaryPhase((ReplicationTask) task, request, channel).run();
+            ReplicationTask replicationTask = (ReplicationTask) task;
+            boolean success = false;
+            PrimaryShardReference primaryShardReference = getPrimaryShardReference(request.shardId());
+            try {
+                if (primaryShardReference.isRelocated()) {
```

Comment:
my reasoning here that this is part of "finding the currently active primary IndexShard". As such is part of the glue layers and not part of the actually operation